.github/workflows/build-and-test.yml

```diff
@@ -0,0 +1,289 @@
+name: test-and-build 
+on:
+  push:
+
+env:
+  TEST_RESULTS: /tmp/test-results # path to where test results are saved
+  CONSUL_VERSION: 1.11.2 # Consul's OSS version to use in tests
+  CONSUL_ENT_VERSION: 1.11.2+ent # Consul's enterprise version to use in tests
+  GOTESTSUM_VERSION: 1.6.4 # You cannot use environment variables with workflows. The gotestsum version is hardcoded in the reusable workflows too.
+
+jobs:
+  validate-helm-gen:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      
+      - name: Setup go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.17.2
+
+      - name: Setup go mod cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cache/go-build
+            ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+
+      - name: Validate helm gen 
+        working-directory: hack/helm-reference-gen 
+        run: |
+          go run ./... -validate
+
+  go-fmt-vet-helm-gen:
+    uses: hashicorp/consul-k8s/.github/workflows/reusable-go-fmt-vet.yml@crt-build-and-tests
+    with:
+      directory: hack/helm-reference-gen
+      go-version: 1.17.2
+
+  unit-helm-gen:
+    needs: [go-fmt-vet-helm-gen, validate-helm-gen]
+    uses: hashicorp/consul-k8s/.github/workflows/reusable-unit.yml@crt-build-and-tests
+    with:
+      directory: hack/helm-reference-gen
+      go-version: 1.17.2
+
+  unit-test-helm-templates:
+    needs: [unit-helm-gen]
+    runs-on: ubuntu-latest
+    container: 
+      image: docker.mirror.hashicorp.services/hashicorpdev/consul-helm-test:0.10.0
+      options: --user root
```

Comment:
I'm curious about this one - do we need to be root to run these tests?

Reply:
Good catch. I changed it to user 1001.